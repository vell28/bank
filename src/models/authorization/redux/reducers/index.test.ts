import { authorization, IAuthorizationState } from '.';
import {
  AUTH_SET_TOKEN, AUTH_UPDATE_TOKEN, AUTH_REMOVE_TOKEN, TOGGLE_LOGOUT_CONFIRM_MODAL
} from '../actions';

const initToken: IAuthorizationState = {
  isLoading: false,
  isSuccess: false,
  isError: false,
  data: {},
  errors: [],
};

const data = {
  service_code: 'OTFIN',
  phone: '79111111111',
  token: '71ce2e5123e86619d033722a205508df',
  expires_in: 2592000,
  isShownLogoutConfirmModal: true,
};

const updateData = {
  success: true,
  token: '9bb5da20816e1e42a3a9a524cb7573a9',
};

describe('authorization reducers', () => {
  test('token_reducer/AUTH_SET_TOKEN', () => {
    const action = {
      type: `${AUTH_SET_TOKEN}_SUCCESS`,
      payload: {
        data,
      },
    };

    const newState = authorization(initToken, action);
    expect(newState.data.token).toBe(data.token);
    expect(newState.isSuccess).toBe(true);
  });

  test('token_reducer/AUTH_UPDATE_TOKEN', () => {
    const action = {
      type: `${AUTH_UPDATE_TOKEN}_SUCCESS`,
      payload: {
        data: updateData,
      },
    };

    const newState = authorization(initToken, action);
    expect(newState.data.token).toBe(updateData.token);
    expect(newState.isSuccess).toBe(true);
    expect(newState).toMatchSnapshot();
  });

  test('token_reducer/AUTH_REMOVE_TOKEN', () => {
    const action = {
      type: AUTH_REMOVE_TOKEN,
    };

    const newState = authorization(initToken, action);
    expect(newState.data).toEqual({
      service_code: 'OTFIN',
      'dev-id': 'test',
      'token-firebase': '',
      isShownLogoutConfirmModal: false,
    });
    expect(newState).toMatchSnapshot();
  });
});

test('authorization/TOGGLE_LOGOUT_CONFIRM_MODAL', () => {
  const action = {
    type: TOGGLE_LOGOUT_CONFIRM_MODAL,
    payload: false,
  };

  const newState = authorization(initToken, action);
  expect(newState.data.isShownLogoutConfirmModal).toBeFalsy();
});
