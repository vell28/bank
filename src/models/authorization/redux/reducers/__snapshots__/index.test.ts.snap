// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authorization reducers token_reducer/AUTH_REMOVE_TOKEN 1`] = `
Object {
  "data": Object {
    "dev-id": "test",
    "isShownLogoutConfirmModal": false,
    "service_code": "OTFIN",
    "token-firebase": "",
  },
  "errors": Array [],
  "isError": false,
  "isLoading": false,
  "isSuccess": false,
}
`;

exports[`authorization reducers token_reducer/AUTH_UPDATE_TOKEN 1`] = `
Object {
  "data": Object {
    "success": true,
    "token": "9bb5da20816e1e42a3a9a524cb7573a9",
  },
  "errors": Array [],
  "isError": false,
  "isLoading": false,
  "isSuccess": true,
}
`;
